Cleaning

yannick1974 opened the pull request:
Move the TNG100 dataset to code/datasets.

(I also reformatted the code with black).

yannick1974 committed:
refactor: Move datasets to code/datasets

yannick1974 committed:
refactor(dataset): Reformat code

yannick1974 committed:
refactor(datasets): Use environment variable for data location

EiffL commented:
oui 'cest pas mal, mais je pense qu'il vaut mieux retirer tout l'ancien dossier sfh dans ce cas, plutot que de retirer uniguement le fichier tng100.py

EiffL commented:
et meme du coup your folder "code" could even be named `sfh`, and then  you would be able to do:
```
from sfh.datasets import tng100
```

yannick1974 committed:
fix: Make stuff installable

yannick1974 committed:
chore: Add .envrc to gitignore

yannick1974 committed:
refactor: Move old code to legacy directory

yannick1974 committed:
refactor: Rename module to sfh old name

yannick1974 committed:
feat: Add function to set tensorflow_datasets environment

With it, it is possible to do:

```python
from sfh.datasets import setup_environment

setup_environment()
```

so that tensorflow_datasets finds the datasets without having to do
symlinks in the home directory.

yannick1974 committed:
docs: Add documentation

yannick1974 committed:
docs: Fix documentation...

yannick1974 commented:
J'ai renommé le module en `sfh` (je l'avais appelé `hackathon`). J'ai aussi rajouté une petite fonction pour positionner correctement le `data_dir` dir `tensorflow_datasets` sans avoir à jouer avec les liens symboliques.

J'ai aussi déplacé l'ancien code dans le répertoire `legacy`.

yannick1974 commented:
Par contre, j'ai cette erreur quand j'essaie de charger les datasets sur Jean Zay:
```
ParseError: Failed to construct dataset tng100: Message type "tensorflow_datasets.DatasetInfo" has no field named "fileFormat".
 Available Fields(except extensions): ['name', 'description', 'version', 'configName', 'configDescription', 'citation', 'sizeInBytes', 'downloadSize', 'location', 'downloadChecksums', 'schema', 'splits', 'supervisedKeys', 'redistributionInfo']
```

yannick1974 committed:
refactor(datasets): Make dataset downloadable

Try to combine the use on Jean Zay with the TNG100 dataset present in
the share folder and the downloading of the dataset.

TODO: Test when we have a link.

yannick1974 commented:
Oublie l'erreur ci-dessus. C'est que je ne chargeais pas le bon environnement tensorflow. ;-)
